src/Player/Player.cpp

```diff
@@ -45,46 +45,202 @@ Player::~Player() {
+        return false;
+    }
+
+    int spendableArmies;
+    std::string regionName;
+    std::vector<std::string> placementRegions;
+
+    // if player doesn't have enough armies to fulfill the value on card, place remaining armies
+    if (totalArmies > *armies) {
+        spendableArmies = *armies;
+    }
+    else {
+        spendableArmies = totalArmies;
+    }
+
+    // inform player where they can place units (ie. starting area and where they have cities)
+    if (spendableArmies == 1) {
+        std::cout << *name << ", you have " << spendableArmies << " army to place on the map." << endl;
+    }
+    else {
+        std::cout << *name << ", you have " << spendableArmies << " armies to place on the map." << endl;
+    }
+    std::cout << "Here are the areas in which you may place an army: " << endl;
+    placementRegions = gameBoard.getRegionsToAddArmies(*name);
+    for (int i = 0; i < placementRegions.size(); i++) {
+        std::cout << "- " << placementRegions.at(i) << endl;
+    }
+
+    // place new armies
+    for (int i = 0; i < spendableArmies; i++) {
+        std::cout << "Where would you like to place your army? " << endl;
+        regionName = validateRegion(placementRegions);
+        (*armies)--;
+        gameBoard.addArmy(regionName, *name);
+        std::cout << "Successfully added an army to " << regionName << "." << endl << endl;
+    }
+
+    return true;
 }
 
-// TODO: Assignment 2
-void Player::moveOverWater(int totalMoves) {
-    std::cout << "This method will allow the player to move " << totalMoves << " armies across water." << std::endl;
+void Player::moveArmies(int totalMoves, Map &gameBoard, bool waterMove)
+{
+    for (int i = 0; i < totalMoves; i++) {
+        std::string origin;
+        std::string destination;
+        std::vector<std::string> regionsWithArmies;
+        std::vector<std::string> connections;
+
+        // display regions where player currently has armies
+        std::cout << *name << ", you have armies in the following locations: " << endl;
+        regionsWithArmies = gameBoard.getRegionsWithArmies(*name);
+        for (int i = 0; i < regionsWithArmies.size(); i++) {
+            std::cout << "- " << regionsWithArmies.at(i) << endl;
+        }
+
+        // player selects a region to move an army from
+        std::cout << "Please select a region in which to move an army from: ";
+        origin = validateRegion(regionsWithArmies);
+
+        // display connected regions
+        std::cout << "The follow regions are connected to " << origin << " by land: " << endl;
+        if (waterMove) {
+            connections = gameBoard.getRegionsConnectedByLandAndWater(origin);
+        }
+        else {
+            connections = gameBoard.getRegionsConnectedByLand(origin);
+        }
+        for (int i = 0; i < connections.size(); i++) {
+            std::cout << "- " << connections.at(i) << endl;
+        }
```

Comment:
Does this mean that if it's a waterMove, it'll always say it's connected by land?

Reply:
woops. yeah it shouldnt say that